### PR TITLE
change IBlock's implementation

### PR DESCRIPTION
**Pull request by friendlyhj**

### CODE REVIEW

File: CraftTweaker2-API/src/main/java/crafttweaker/api/block/IBlock.java [hunk 1]
```diff
@@ -27,7 +28,12 @@
...
     @ZenGetter("data")
     IData getTileData();
-    
+
     @ZenGetter("fluid")
     ILiquidDefinition getFluid();
+
+    default IData getTileDataInternal() {
+        CraftTweakerAPI.logError("Class " + this.getClass().getCanonicalName() + " doesn't override IBlock::getTileDataInternal! It is a bug!");
+        return null;
```
Comment (kindlich) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/block/IBlock.java]: You could return here `getTileData()` as fallback

File: CraftTweaker2-API/src/main/java/crafttweaker/api/block/IBlock.java [hunk 2]
```diff
@@ -27,7 +28,12 @@
     
     @ZenGetter("data")
     IData getTileData();
-    
+
     @ZenGetter("fluid")
     ILiquidDefinition getFluid();
+
+    default IData getTileDataInternal() {
```
Comment (kindlich) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/block/IBlock.java]: What exactly do we need a new method for here?

**Review by kindlich — COMMENTED** [review]
I don't quite understand why we'd need that new method.
Could you give me a usecase, like what issue does it fix?

**Comment by jaredlll08** [comment]
Yea what does this PR actually do?

**Comment by friendlyhj** [comment]
This PR can send a warning to people who wants to use `world.getBlockState(pos).block.data` to get Block NBT. (For example, #989) But as you know, it always return null. The problem had doubted me and many people for a long time. Although I made warning in documentation in previous PR, I think making it in code is better. Because blockstates cannot storage information more than 4 bits, we should use tile entity, which people who never read Minecraft code don't know. In fact, I think `block#getTileData` is a poor idea. Blocks don't have data, the tile entity do. There should be a `TileEntity` zenclass in CraftTweaker.

I added the new method to avoid warnings in internal calling `getTileData` method. Yea, since I changed PlayerInteractEvent's implementation, the `getBlock` method will return `MCWorldBlock`. So the new method is useless in a sense. I will delete it later.

**Review by kindlich — APPROVED** [review]
Looks good to me

**Pull request merged**